contracts/libs/TokenRouter.sol

```diff
@@ -52,16 +52,19 @@ abstract contract TokenRouter is Router {
         bytes32 _recipient,
         uint256 _amount
     ) external payable {
-        _transferFromSender(_amount);
+        bytes memory metadata = _transferFromSender(_amount);
         _dispatchWithGas(
             _destination,
-            Message.format(_recipient, _amount),
+            Message.format(_recipient, _amount, metadata),
             msg.value
         );
         emit SentTransferRemote(_destination, _recipient, _amount);
     }
 
-    function _transferFromSender(uint256 _amount) internal virtual;
+    function _transferFromSender(uint256 _amount)
```

Comment:
I think Natspecs for these function that are supposed to be override would be very useful